src/types.ts

```diff
@@ -164,16 +170,23 @@ export type TurboAuthenticatedClientConfiguration = {
 
 export type FileStreamFactory =
   | (() => Readable)
-  | (() => ReadableStream)
+  | WebFileStreamFactory
   | (() => Buffer);
+
+export type WebFileStreamFactory = () => ReadableStream;
+
 export type SignedDataStreamFactory = FileStreamFactory;
 export type StreamSizeFactory = () => number;
-export type TurboFileFactory = {
-  fileStreamFactory: FileStreamFactory; // TODO: allow multiple files
+export type TurboFileFactory<T = FileStreamFactory> = {
+  fileStreamFactory: T; // TODO: allow multiple files
   fileSizeFactory: StreamSizeFactory;
+  opts?: DataItemOptions;
```

Comment:
I think I'd like to see this as `dataItemOpts` across the codebase. Perhaps there might be other `opts` that are relevant in a minor, backwards-compatible update to the SDK, e.g. retry opts, gateway opts, chunking/batching opts, etc. `opts` is just too general.

Reply:
You can remove the TODO on line 268 that reads:
```
export interface TurboAuthenticatedUploadServiceInterface
  extends TurboUnauthenticatedUploadServiceInterface {
  // TODO: support target, anchor and tags
  uploadFile({
```

Reply:
b428a15